src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +158,382 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
-      this.deselect(previouslySelected);
-    }
-
-    option.selected ? this._selectionModel.select(option) : this._selectionModel.deselect(option);
+  /**
+   * Registers a callback to be invoked when the listbox's value changes from user input.
+   * @param fn The callback to register
+   * @docs-private
+   */
+  registerOnChange(fn: (value: T[]) => void): void {
+    this._onChange = fn;
   }
 
-  _updatePanelForSelection(option: CdkOption<T>) {
-    if (this._combobox) {
-      if (!this.multiple) {
-        this._combobox.updateAndClose(option.selected ? option.value : []);
-      } else {
-        this._combobox.updateAndClose(this.getSelectedValues());
-      }
-    }
+  /**
+   * Registers a callback to be invoked when the listbox is blurred by the user.
+   * @param fn The callback to register
+   * @docs-private
+   */
+  registerOnTouched(fn: () => {}): void {
+    this._onTouched = fn;
   }
 
-  /** Toggles the selected state of the active option if not disabled. */
-  private _toggleActiveOption() {
-    const activeOption = this._listKeyManager.activeItem;
-    if (activeOption && !activeOption.disabled) {
-      activeOption.toggle();
+  /**
+   * Sets the listbox's value.
+   * @param value The new value of the listbox
+   * @docs-private
+   */
+  writeValue(value: T[]): void {
+    if (this.options) {
+      this.selectionModel().setSelection(...(value == null ? [] : coerceArray(value)));
     }
   }
 
-  /** Returns the id of the active option if active descendant is being used. */
-  _getAriaActiveDescendant(): string | null | undefined {
-    return this._useActiveDescendant ? this._listKeyManager?.activeItem?.id : null;
+  /**
+   * Sets the disabled state of the listbox.
+   * @param isDisabled The new disabled state
+   * @docs-private
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
   }
 
-  /** Updates the activeOption and the active and focus properties of the option. */
-  private _updateActiveOption() {
-    if (!this._listKeyManager.activeItem) {
-      return;
-    }
-
-    this._activeOption?.deactivate();
-    this._activeOption = this._listKeyManager.activeItem;
-    this._activeOption.activate();
+  /** The selection model used to track the listbox's value. */
+  protected selectionModel() {
+    return this.selectionModelSubject.value;
+  }
 
-    if (!this.useActiveDescendant) {
-      this._activeOption.focus();
+  /**
+   * Triggers the given option in response to user interaction.
+   * - In single selection mode: selects the option and deselects any other selected option.
+   * - In multi selection mode: toggles the selected state of the option.
+   * @param option The option to trigger
+   */
+  protected triggerOption(option: CdkOption<T> | null) {
+    if (option && !option.disabled) {
+      let changed = false;
+      const subscription = this.selectionModel().changed.subscribe(() => (changed = true));
```

Comment:
It might be better to use a `take(1)` here instead of unsubcribing from inside the `subscribe`. I remember there being some issues with this if people use the package that patches `rxjs` with zone.js.